featuretools/tests/conftest.py

```diff
@@ -156,6 +156,66 @@ def ks_diamond_es(pd_diamond_es):
     return ft.EntitySet(id=pd_diamond_es.id, entities=entities, relationships=relationships)
 
 
+@pytest.fixture(params=['pd_transactions_defaultvalue_es', 'dask_transactions_defaultvalue_es', 'ks_transactions_defaultvalue_es'])
+def transactions_defaultvalue_es(request):
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture
+def pd_transactions_defaultvalue_es():
+    transactions = pd.DataFrame({
+        "id": [1, 2, 3, 4],
+        "session_id": ["a", "a", "b", "c"],
+        "value": [1, 1, 1, 1]
+    })
+
+    sessions = pd.DataFrame({
+        "id": ["a", "b"]
+    })
+
+    es = ft.EntitySet()
+    es.entity_from_dataframe(entity_id="transactions",
+                             dataframe=transactions,
+                             index="id")
+    es.entity_from_dataframe(entity_id="sessions",
+                             dataframe=sessions,
+                             index="id")
+
+    es.add_relationship(ft.Relationship(es["sessions"]["id"], es["transactions"]["session_id"]))
+    return es
+
+
+@pytest.fixture
+def dask_transactions_defaultvalue_es(pd_transactions_defaultvalue_es):
+    entities = {}
+    for entity in pd_transactions_defaultvalue_es.entities:
+        entities[entity.id] = (dd.from_pandas(entity.df.reset_index(), npartitions=4), entity.index, None, entity.variable_types)
```

Comment:
Also, I don't think you need the `.reset_index()` call here. The entity.df does not appear to have the index set, so calling reset_index here is giving an extra column named `index` that we don't need for anything.

Reply:
The index information was lost even without the `.reset_index()` call.

Reply:
I guess maybe I don't understand what you mean by the index information being lost - we typically do not rely on any information in the dataframe index for Dask or Koalas dataframes, but instead we use a column in the dataframe that is defined to be the index - in this case `id`. However, when we create a feature matrix from a pandas dataframe, the feature matrix dataframe has its index set to the index column. For Dask and Koalas the dataframe index is not set (so the index values are not meaningful), but rather there is a normal column in the dataframe that contains the index values.

Reply:
Because of this difference, and the fact that we can't rely on the row order in our Dask or Koalas feature matrix, we have the ability to set the index and sort the index via our `to_pandas` test utility function.

As I was trying to understand this issue, I was able to get `test_calculate_feature_matrix_returns_default_values` to pass with pandas, Dask and Koalas inputs if I removed the `reset_index` call here along with making the other changes I mentioned in `test_calculate_feature_matrix.py`.

Are you still seeing problems after making those changes? Maybe I am missing something still.

Reply:
This fixed the issue for me: `feature_matrix = to_pandas(feature_matrix, index='id', sort_index=True)`
@thehomebrewnerd thanks for your patience, its my first time working with dask/koalas

Reply:
No problem - I know it can be a little confusing because of the different ways things are handled between pandas/Dask and Koalas. Glad to hear that fixed the issue!

Reply:
@thehomebrewnerd Quick question a bit off-topic: why does CircleCI ask me access to my private repos even to check the status of my CI/CD jobs for this repo?

Reply:
@seriallazer Not sure exactly - are you seeing that as part of the process of authenticating with Github? 

@rwedge might know a bit more of the details there than I do.

Reply:
@thehomebrewnerd yes part of the Github authentication.

Also most of the tests have passed but the CircleCI status shows "Unauthorized". Not really sure whats the process for CircleCI honestly (couldnt find any documentation about it).

Reply:
@seriallazer I believe it's due to this being a forked PR.  We're planning to switch to a different CI process soon, so we could wait for that.  An alternative we could do is make a copy of your branch on this repo and trigger a test build on that version of the branch.

Reply:
@rwedge I see, sure I can do that going forward.
Although for this PR, the status still says "Unauthorized" even though all tests pass now.
Maybe someone with appropriate perms can retrigger the full-set?